### PR TITLE
Add devise for User Authentication

### PR DESCRIPTION
**Pull request by Wils-FOURR-speed**
### Description:
This pull request addresses the project requirements by implementing Devise for user authentication in the blog app. The changes include:

- Removal of the current_user method from the ApplicationController as Devise will provide this functionality.
- Installation and setup of Devise to handle user registration, login, email confirmation, and password reset.
- Modification of Devise views for registration.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      <div class="actions my-5">
+        <%= f.submit "Sign up", class: 'border bg-black text-white px-3 py-1 rounded-md' %>
+      </div>
+
+      <%= render "devise/shared/links" %>
+
+    <% end %>
+  </div>
+</div>
```
Comment (Munsa1) [inline app/views/devise/registrations/new.html.erb]: - [ ] Great job on your app really. Kindly style the login and signup buttons to span the width of the input fields as is in the given wireframe

Given frame
image

Your design
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Wils-FOURR-speed, 
required changes

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Implemented Authentication :heavy_check_mark: 
Nice Styling :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Wils-FOURR-speed  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1:  

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Wils-FOURR-speed** [comment]
Thanks @Lucas-Masaba for this

**Pull request closed without merging**